### PR TITLE
Set version to 1.000, provide version setter script, update composites

### PR DESCRIPTION
**Pull request by madig**
The baseline release is supposed to match the Google Fonts release as
closely as possible. Since the GF release has been used in the wild for
years, it is appropriate to set the version to 1.000 and then go from
there.

The version script changes the metadata as per the recommendations in
http://silnrsi.github.io/FDBP/en-US/Versioning.html and additionally
updates the version glyph (uniEFFD).

Since the version glyph was previously inconsistent and contained
components and sometimes decomposed ones, I had to update the composite
information with `python -m vttLib dump font.ttf font.ufo` before VTT
compilation succeeded. Due to spacing changes, there is a large amount
of churn.

Question: the UniqueIDs are all changed slightly, e.g. "Monospaced" instead of "Mono". That okay?

### CODE REVIEW

File: source/Ubuntu-B.ufo/data/com.github.fonttools.ttx/T_S_I__0.ttx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<ttFont sfntVersion="\x00\x01\x00\x00" ttLibVersion="3.0">
+<ttFont sfntVersion="\x00\x01\x00\x00" ttLibVersion="3.21">
```
Comment (sladen) [inline source/Ubuntu-B.ufo/data/com.github.fonttools.ttx/T_S_I__0.ttx]: The `ttLibVersion` nump is diff noise, and ideally should be filtered out from the actual commit.

File: source/Ubuntu-B.ufo/data/com.github.fonttools.ttx/T_S_I__1.ttx [hunk 2]
```diff
@@ -194,8 +194,8 @@ IUP[Y]
 /* Composite glyph autohinted */
 
 USEMYMETRICS[]
-OFFSET[R],161,0,0
-OFFSET[R],143,485,160
+OFFSET[R], 161, 0, 0
+OFFSET[R], 143, 485, 160
```
Comment (sladen) [inline source/Ubuntu-B.ufo/data/com.github.fonttools.ttx/T_S_I__1.ttx]: The `OFFSET[R]` whitespace changes are diff noise, and ideally should be filtered out from the actual commit.

File: source/Ubuntu-R.ufo/fontinfo.plist [hunk 3]
```diff
@@ -107,9 +107,9 @@ The scope of the Ubuntu Font Family includes all the languages used by the vario
 		<array>
 		</array>
 		<key>openTypeNameUniqueID</key>
-		<string>Ubuntu Regular Version 0.83</string>
+		<string>Ubuntu Regular Version 1.000</string>
 		<key>openTypeNameVersion</key>
-		<string>Version 0.83</string>
+		<string>Version 1.000</string>
```
Comment (sladen) [inline source/Ubuntu-R.ufo/fontinfo.plist]: The addition of the *code* to auto-update the version number should be completely separate from the proposal to bump the version number to `1.0`.  For the moment it should still generate `0.83`.

(Switching to a higher number should be the *absolutely very last change before making a release*.  This has caused so many problems in the past.

File: source/Ubuntu-R.ufo/fontinfo.plist [hunk 4]
```diff
@@ -267,9 +267,9 @@ The scope of the Ubuntu Font Family includes all the languages used by the vario
 		<key>unitsPerEm</key>
 		<integer>1000</integer>
 		<key>versionMajor</key>
-		<integer>0</integer>
+		<integer>1</integer>
 		<key>versionMinor</key>
-		<integer>830</integer>
+		<integer>0</integer>
```
Comment (sladen) [inline source/Ubuntu-R.ufo/fontinfo.plist]: This version changing should be in a separate commit.  It should probably match consistently and be `000`

File: source/Ubuntu-R.ufo/glyphs/uniE_F_F_D_.glif [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
-		<component base="zerosuperior" yOffset="-313"/>
-		<component base="period" xOffset="359"/>
-		<component base="eightsuperior" xOffset="605" yOffset="-313"/>
-		<component base="zerosuperior" xOffset="964" yOffset="-313"/>
-		<component base="zerosuperior" xOffset="1323" yOffset="-313"/>
+		<component base="one"/>
+		<component base="period" xOffset="564"/>
+		<component base="zero" xOffset="810"/>
+		<component base="zero" xOffset="1374"/>
+		<component base="zero" xOffset="1938"/>
```
Comment (sladen) [inline source/Ubuntu-R.ufo/glyphs/uniE_F_F_D_.glif]: This debug glyph is used for double-checking user-provided screenshots during debugging.  Instead of copying the debug glyph some users have actually typed out the version number.  The use of the smaller `superior` figures allows telling apart a screenshot with a real debug symbol, against a manually typed one.

It should be possible to reduce the diff noise by retaining the `yOffset` and `xOffset` as they were.  Ideally any versioning change should be a separate commit from the diff noise.

File: source/Ubuntu-RI.ufo/fontinfo.plist [hunk 6]
```diff
@@ -95,9 +95,9 @@ The scope of the Ubuntu Font Family includes all the languages used by the vario
 		<array>
 		</array>
 		<key>openTypeNameUniqueID</key>
-		<string>Ubuntu Regular Version 0.83</string>
+		<string>Ubuntu Italic Version 1.000</string>
```
Comment (sladen) [inline source/Ubuntu-RI.ufo/fontinfo.plist]: Change in `Ubuntu-RI.ufo` from "Ubuntu Regular Version" to "Ubuntu Italic Version".  This is probably useful and correct, but should be a separate, clearly documented pull request, from the changing of the version number.

File: source/Ubuntu-R.ufo/fontinfo.plist (hunk 4 above)
Comment (madig) [reply source/Ubuntu-R.ufo/fontinfo.plist]: Not sure how to do that since these are really just ints internally. Glyphs displays it as "000" though.

File: source/Ubuntu-B.ufo/data/com.github.fonttools.ttx/T_S_I__0.ttx (hunk 1 above)
Comment (anthrotype) [reply source/Ubuntu-B.ufo/data/com.github.fonttools.ttx/T_S_I__0.ttx]: I disagree. It’s important to track the fonttools version used to dump a ttx file as new attributes or elements may be added over time to the format, and parsers may take that into account

Comment (sladen) [reply source/Ubuntu-B.ufo/data/com.github.fonttools.ttx/T_S_I__0.ttx]: Yes, in a *separate commit* that clearly states the origin of the change.  ie. that the change is fully understood and not diff noise that confuses the true purpose of the improvement being proposed.

**Comment by madig** [comment]
(seperating into different PRs)

File: source/Ubuntu-R.ufo/glyphs/uniE_F_F_D_.glif (hunk 5 above)
Comment (madig) [reply source/Ubuntu-R.ufo/glyphs/uniE_F_F_D_.glif]: yOffset is tossed because I'm using inferiors instead of shifting superiors.

File: source/Ubuntu-RI.ufo/fontinfo.plist (hunk 6 above)
Comment (madig) [reply source/Ubuntu-RI.ufo/fontinfo.plist]: Why not one "Set version metadata to 0.830 UFO Source" that fixes these and adds updated components to everything in one go?
